### PR TITLE
Support Xcode13

### PR DESCRIPTION
**Pull request by funzin**
## Overview
- Support Xcode13(Swift5.3 → Swift5.5)
- Update dependencies in case of Swift5.5
  - https://github.com/apple/swift-package-manager.git
  - https://github.com/apple/swift-argument-parser.git
  - https://github.com/apple/swift-tools-support-core.git
- Fix error and warning codes

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -3,6 +3,17 @@
...
+let dependencies: [Package.Dependency]
+#if swift(>=5.5)
+dependencies = [.package(url: "https://github.com/apple/swift-argument-parser.git", .exact("0.4.4")),
+                .package(name: "SwiftPM", url: "https://github.com/apple/swift-package-manager.git", .branch("release/5.5")),
+                .package(url: "https://github.com/apple/swift-tools-support-core.git", .exact("0.2.3"))]
+#else
+dependencies = [.package(url: "https://github.com/apple/swift-argument-parser.git", .exact("0.3.2")),
+                .package(name: "SwiftPM", url: "https://github.com/apple/swift-package-manager.git", .revision("swift-5.3.3-RELEASE")),
+                .package(url: "https://github.com/apple/swift-tools-support-core.git", .exact("0.1.12"))]
+#endif
```
Comment (funzin) [inline Package.swift:15]: CI executes build on Xcode12 so separate dependencies by swift version

**Comment by fotiDim** [comment]
@funzin I tried your branch. It does compile but `swift create-xcframework` or `swift create-xcframework --list-products` seem to get stuck and do not output anything when run at the root folder of a swift package. Not sure if it related to your PR since this is the first time I install it.

**Comment by funzin** [comment]
Thanks for your reply!
I will confirm the incorrect behavior🙏

**Comment by fotiDim** [comment]
@funzin yes it seems to proceed now. I stumbled upon another error but it ισ probably unrelated to this PR.

**Comment by funzin** [comment]
@fotiDim 
Umm, I checked created xcframwork🙂
Can you show me your `Package.swift`?

If you create Package.swift like https://github.com/funzin/swift-create-xcframework-example, you may confirm created xcframework.

File: Package.swift (hunk 1 above)
Comment (bok-) [reply Package.swift:15]:
```suggestion
#if swift(>=5.5)
dependencies = [
    .package(url: "https://github.com/apple/swift-argument-parser.git", .exact("0.4.4")),
    .package(name: "SwiftPM", url: "https://github.com/apple/swift-package-manager.git", .branch("release/5.5")),
    .package(url: "https://github.com/apple/swift-tools-support-core.git", .exact("0.2.3"))
]
#else
dependencies = [
    .package(url: "https://github.com/apple/swift-argument-parser.git", .exact("0.3.2")),
    .package(name: "SwiftPM", url: "https://github.com/apple/swift-package-manager.git", .revision("swift-5.3.3-RELEASE")),
    .package(url: "https://github.com/apple/swift-tools-support-core.git", .exact("0.1.12"))
]
#endif
```

File: Sources/CreateXCFramework/PackageInfo.swift [hunk 2]
```diff
@@ -70,17 +70,33 @@ struct PackageInfo {
 
         self.toolchain = try UserToolchain(destination: try .hostDestination())
 
-        let resources = try UserManifestResources(swiftCompiler: self.toolchain.swiftCompiler)
+        let resources: UserManifestResources
+        #if swift(>=5.5)
+            resources = try UserManifestResources(swiftCompiler: self.toolchain.swiftCompiler, swiftCompilerFlags: [])
+        #else
+            resources = try UserManifestResources(swiftCompiler: self.toolchain.swiftCompiler)
+        #endif
```
Comment (bok-) [inline Sources/CreateXCFramework/PackageInfo.swift:78]:
```suggestion
        #if swift(>=5.5)
        let resources = try UserManifestResources(swiftCompiler: self.toolchain.swiftCompiler, swiftCompilerFlags: [])
        #else
        let resources = try UserManifestResources(swiftCompiler: self.toolchain.swiftCompiler)
        #endif

```

File: Sources/CreateXCFramework/PackageInfo.swift [hunk 3]
```diff
@@ -70,17 +70,33 @@ struct PackageInfo {
...
+                                                                            completion: $0)
+            }
+        #else
+            self.graph = self.workspace.loadPackageGraph(root: root, diagnostics: self.diagnostics)
+            self.manifest = try ManifestLoader.loadManifest (
+                packagePath: root,
+                swiftCompiler: self.toolchain.swiftCompiler,
+                packageKind: .root
+            )
+        #endif
```
Comment (bok-) [inline Sources/CreateXCFramework/PackageInfo.swift:99]:
```suggestion
        #if swift(>=5.5)
        self.graph = try self.workspace.loadPackageGraph(rootPath: root, diagnostics: self.diagnostics)
        let swiftCompiler = toolchain.swiftCompiler
        self.manifest = try tsc_await {
            ManifestLoader.loadRootManifest(
                at: root,
                swiftCompiler: swiftCompiler,
                swiftCompilerFlags: [],
                identityResolver: DefaultIdentityResolver(),
                on: DispatchQueue.global(qos: .background),
                completion: $0
            )
        }
        #else
        self.graph = self.workspace.loadPackageGraph(root: root, diagnostics: self.diagnostics)
        self.manifest = try ManifestLoader.loadManifest (
            packagePath: root,
            swiftCompiler: self.toolchain.swiftCompiler,
            packageKind: .root
        )
        #endif
```

File: Sources/CreateXCFramework/PackageInfo.swift [hunk 4]
```diff
@@ -214,7 +230,7 @@ enum SupportedPlatforms {
     case packageValid (plan: [SupportedPlatform])
 }
 
-extension SupportedPlatform: Equatable, Comparable {
+extension SupportedPlatform: Comparable {
```
Comment (bok-) [inline Sources/CreateXCFramework/PackageInfo.swift:233]: This change isn't necessary. Please revert it.

**Review by bok- — COMMENTED** [review]
Hi @funzin. Thanks very much for taking the time to raise this PR and test it out so thoroughly. I've tested it with Xcode 12.5 and Xcode 13.0 for building some project types here and it everything looks good.

I've included some suggestions to align your contributions with the coding style of the rest of the project. If you can update the style to match I'd be happy to approve this PR.

**Comment by funzin** [comment]
@bok- 
Thank you for your review.
I fixed some indents.
f5a528a

**Pull request merged**